In [1]:
QTD_OF_SUMMARIES = 10
NR_EPOCHS = 50
BATCH_SIZE = 100

In [2]:
import datautils.dataloader as loader

summaries = loader.load_summaries()

In [ ]:
loader.load_mne_data(summaries[:QTD_OF_SUMMARIES])
loader.load_segmented_data(summaries[:QTD_OF_SUMMARIES])

### Data Preparation -> Dominio do Tempo 

- Segmentar as informações de tempo em janelas
- Criar as labels para cada janela
- Separar as informações em Treino e Teste
- Alimentar a rede e verificar a saída

In [4]:
len(summaries[0].signal.time_segments)

720

In [5]:
import datautils.splitter.mnesplitter as mnesplitter

X_train, X_val, y_train, y_val = mnesplitter.segmented_time_data_splitter(summaries[:QTD_OF_SUMMARIES])

In [6]:
import ia.model.cnn as ia

time_cnn = ia.CNNModel(input_shape=(X_train[0].shape[0], X_train[0].shape[1]))

In [7]:
time_cnn.compile()

In [8]:
time_cnn.fit(X_train, y_train, num_epochs=NR_EPOCHS, batch_size=BATCH_SIZE, val_data=X_val, val_labels=y_val)

Epoch 1/1000
144/144 [==============================] - 1s 4ms/step - loss: 0.1529 - accuracy: 0.9917 - val_loss: 0.0028 - val_accuracy: 1.0000
Epoch 2/1000
144/144 [==============================] - 0s 3ms/step - loss: 0.0112 - accuracy: 0.9986 - val_loss: 0.0013 - val_accuracy: 1.0000
Epoch 3/1000
144/144 [==============================] - 0s 3ms/step - loss: 0.0109 - accuracy: 0.9986 - val_loss: 4.7135e-04 - val_accuracy: 1.0000
Epoch 4/1000
144/144 [==============================] - 0s 3ms/step - loss: 0.0115 - accuracy: 0.9986 - val_loss: 8.9913e-04 - val_accuracy: 1.0000
Epoch 5/1000
144/144 [==============================] - 0s 3ms/step - loss: 0.0114 - accuracy: 0.9986 - val_loss: 0.0014 - val_accuracy: 1.0000
Epoch 6/1000
144/144 [==============================] - 0s 3ms/step - loss: 0.0108 - accuracy: 0.9986 - val_loss: 5.5724e-04 - val_accuracy: 1.0000
Epoch 7/1000
144/144 [==============================] - 0s 3ms/step - loss: 0.0110 - accuracy: 0.9986 - val_loss: 4.4463e-04

144/144 [==============================] - 1s 4ms/step - loss: 2.4551e-06 - accuracy: 1.0000 - val_loss: 6.7967e-26 - val_accuracy: 1.0000
Epoch 915/1000

144/144 [==============================] - 1s 4ms/step - loss: 0.0354 - accuracy: 0.9986 - val_loss: 1.7228e-07 - val_accuracy: 1.0000
Epoch 916/1000

144/144 [==============================] - 1s 4ms/step - loss: 8.3216e-08 - accuracy: 1.0000 - val_loss: 1.7197e-07 - val_accuracy: 1.0000

In [9]:
# loader.load_mne_data(summaries[QTD_OF_SUMMARIES + 1:QTD_OF_SUMMARIES + 10])
# loader.load_segmented_data(summaries[QTD_OF_SUMMARIES + 1:QTD_OF_SUMMARIES + 10])

In [10]:
# import numpy as np
# X_test = summaries[QTD_OF_SUMMARIES + 1 : QTD_OF_SUMMARIES + 10]
# y_test = [summary.has_anomaly() for summary in X_test]

# X_test_freq_data = np.array([summary.signal.get_freq_data()[0] for summary in X_test])

In [11]:
# predictions = time_cnn.model.predict(X_test_freq_data)
# for i in range(len(y_test)):
#     print(str(predictions[i]) + ":" + str(y_test[i]))
# test_loss, test_acc = frequency_cnn.model.evaluate(X_test_freq_data, y_test)

# print(f'Test accuracy: {test_acc}')